Jenkinsfile

```diff
@@ -0,0 +1,73 @@
+library 'magic-butler-catalogue'
+def PROJECT_NAME = 'logdna-agent-v2'
+def RUST_IMAGE_REPO = 'us.gcr.io/logdna-k8s/rust'
+
+pipeline {
+    agent any
+    options {
+        timestamps()
+        ansiColor 'xterm'
+    }
+    triggers {
+        cron(env.BRANCH_NAME ==~ /\d\.\d/ ? 'H H 1,15 * *' : '')
+    }
+    stages {
+        stage('Test') {
+            steps {
+                sh """
+                    make lint RUST_IMAGE_REPO=${RUST_IMAGE_REPO}
+                    make test RUST_IMAGE_REPO=${RUST_IMAGE_REPO}
+                """
+            }
+            post {
+                success {
+                    sh "make clean RUST_IMAGE_REPO=${RUST_IMAGE_REPO}"
+                }
+            }
+        }
+        stage('Build & Publish Images') {
+            stages {
+                stage('Build Image') {
+                    steps {
+                        sh "make build-image RUST_IMAGE_REPO=${RUST_IMAGE_REPO}"
+                    }
+                }
+                stage('Check Publish Images') {
+                    when {
+                        branch pattern: "\\d\\.\\d", comparator: "REGEXP"
+                    }
+                    stages {
+                        stage('Publish Images') {
+                            input {
+                                message "Should we publish the versioned image?"
+                                ok "Publish image"
+                            }
+                            steps {
+                                withCredentials([string(credentialsId: 'icr-iamapikey', variable: 'ICR_IAMAPIKEY'), string(credentialsId: 'dockerhub-access-token', variable: 'DOCKERHUB_ACCESS_TOKEN')]) {
+                                    sh '''
```

Comment:
we really should build this into the jenkins instance like our other creds, so that this kind of thing doesnt have to exist in multiple places

Reply:
Same comment as above. I would imagine we want to only be logged in when we absolutely need to be to minimize any attack surfaces.

If we want to have the Jenkins slaves to always be logged in to the registries, I can look into it.

Reply:
Its not really minimizing the attack surfaces much. every running job has access to the credentials.

Reply:
I opted to replace this with `docker.withRegistry`